docs/beyond_basics/101-181-metalad.rst

```diff
@@ -4,10 +4,363 @@ Metadata-Management with MetaLad
+The two parts are separated by ``:``. The following line would just dump all metadata for ``zen.png``.
+
+.. runrecord:: _examples/DL-101-181-106
+   :language: console
+   :workdir: beyond_basics/meta/cozy-screensavers
+
+   $ datalad meta-dump -d . .:zen.png
+
+This could also be printed a bit more readable:
+
+.. runrecord:: _examples/DL-101-181-107
+   :language: console
+   :workdir: beyond_basics/meta/cozy-screensavers
+
+   $ datalad -f json_pp meta-dump -d . .:zen.png
+
+.. find-out-more:: More complex metadata-dumps
+
+   TODO: add complex Dataset-file-path-pattern examples, e.g., with UUIDs, versions, etc
+
+Using existing extractors to add metadata
+"""""""""""""""""""""""""""""""""""""""""
+
+TODO: This needs and example
+
+
+Creating your own extractor
+"""""""""""""""""""""""""""
+
+The MetaLad docs have a dedicated user guide that walks you through the process of creating your own extractor. Have a look at `docs.datalad.org/projects/metalad/user_guide/writing-extractors.html <https://docs.datalad.org/projects/metalad/user_guide/writing-extractors.html>`_.
+
+Distributing and Getting Metadata
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Once metadata has been added to a dataset, it can be distributed and retrieved.
+Instead of creating and adding metadata yourself, you could download fitting pre-existing metadata.
+Similarly, instead of repeating a ``meta-add`` process for one and the same files across hierarchies of datasets, metadata added into one dataset can be exported into other datasets.
+Regardless of whether it is a distribution or a retrieval process, though, an export with MetaLad will only concern the *metadata*, and never the primary data.
+
+Download Metadata from a remote repository
+""""""""""""""""""""""""""""""""""""""""""
+
+Let's start by creating a place where someone else's metadata could live.
+
+.. runrecord:: _examples/DL-101-181-110
+   :language: console
+   :workdir: beyond_basics/meta
+
+   $ datalad create metadata-assimilation
+   $ cd metadata-assimilation
+
+Because MetaLad stores metadata in :term:`Git`'s object store, we use Git to directly fetch metadata from a remote repository, such as this demo on :term:`GitHub`: ``https://github.com/christian-monch/metadata-test.git``.
+Because metadata added by MetaLad is not transported automatically but needs to be specifically requested, the command to retrieve it looks unfamiliar to non-Git-users: It identifies the precise location of the :term:`ref` that contains the metadata.
+
+.. find-out-more:: Exactly where is metadata stored?
+   :name: fom-metadataobjecttree
+
+   MetaLad employs an internal metadata model that makes the following properties possible:
+
+   * Metadata has a version encoded, but isn't itself version controlled
+   * Metadata should not be transported if not explicitly requested
+   * It should be possible to only retrieve parts of the overall metadata tree, e.g. certain sub-nodes
+
+   TODO this needs much more
```

Comment:
I still need some insights into the metadata organization. @christian-monch do I see correctly that the metadata is organized internally by "dataset_path"? And without -r, dump would only dump one (the first?) dataset_path, or one that is explicitly specified?

Reply:
That is exactly right!

Here is a longwinded explanation that confirms your view:

Without ``-r`` ``meta-dump`` would only dump metadata for objects specified by the ``DATASET_FILE_PATH_PATTERN``. If ``DATASET_FILE_PATH_PATTERN`` is not provided, an empty string is assumed, which by default translates into a ``tree-file pattern`` (see ``datalad meta-dump --help``) that identifies the topmost superset in the metadata information.

Since there is no studyminimeta-file in the superset, there is also no metadata information and nothing is emitted. But the topmost dataset is still matched by the ``DATASET_FILE_PATH_PATTERN`` and ``-r`` instructs ``meta-dump`` to list metadata for all sub-datasets in the metadata information. So the command line ``datalad meta-dump study-100`` would output metadata because the metadata information contains a dataset at the path "study-100" that has studyminimeta-metadata associated with it.

Reply:
thx!